go/master/etcd_client.go

```diff
@@ -18,6 +18,19 @@ const (
 	DefaultAddrPath = "/master/addr"
 )
 
+// DatabaseOperator is an interface fo database operator,
+// it's useful for unittest
+type DatabaseOperator interface {
+	WaitMasterReady(key string, interval int) []byte
+	WatchWithKey(key string, valChan chan string)
+}
+
+// EtcdClientWithoutLock is the etcd client the master users
+// for service discovery
+type EtcdClientWithoutLock struct {
```

Comment:
Done.
`WaitMasterReady ` => `GetKey`, and I also think this function does not need to be blocked, if the master address is empty, the function `monitorMaster` will not connect to master.